src/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -19,14 +19,28 @@ public static void WriteCyclicFail()
         }
 
         [Fact]
-        [ActiveIssue(37313)]
         public static void WriteTestClassWithArrayOfElementsOfTheSameClassWithoutCyclesDoesNotFail()
         {
             TestClassWithArrayOfElementsOfTheSameClass obj = new TestClassWithArrayOfElementsOfTheSameClass();
 
             //It shouldn't throw when there is no real cycle reference, and just empty object is created
             string json = JsonSerializer.ToString(obj);
-            Assert.Equal(@"{}", json);
+            Assert.Equal(@"{""Array"":null}", json);
+        }
+
+        public class TestClassWithCycle
+        {
+            public TestClassWithCycle Parent { get; set; }
+
+            public void Initialize()
+            {
+                Parent = this;
+            }
+        }
+
+        public class TestClassWithArrayOfElementsOfTheSameClass
+        {
+            public TestClassWithArrayOfElementsOfTheSameClass[] Array { get; set; }
```

Comment:
Can we add a couple more tests for different collection types, like `List<T>` and `Dictionary`, along with one where the recursive property is deeper within a .NET object graph?

Reply:
Sure. Adding a more complex test.